+sw_tests/+unit_tests/unittest_spinw_addmatrix.m

```diff
@@ -0,0 +1,89 @@
+classdef unittest_spinw_addmatrix < matlab.mock.TestCase
+
+    properties
+        swobj = [];
+    end
+
+    methods (TestMethodSetup)
+        function setup_spinw_model(testCase)
+            testCase.swobj = spinw; % default init
+        end
+    end
+
+    methods (Test)
+        function test_no_input_calls_help(testCase)
+            % Tests that if call addmatrix with no input, it calls the help
+            help_function = sw_tests.utilities.mock_function('swhelp');
+            testCase.swobj.addmatrix();
+            testCase.assertEqual(help_function.n_calls, 1);
+            testCase.assertEqual(help_function.arguments, ...
+                {{'spinw.addmatrix'}});
+        end
+        
+        function test_non_numeric_input_warns_and_not_add_matrix(testCase)
+            testCase.verifyWarning(...
+                @() testCase.swobj.addmatrix('value', 'str'), ...
+                'spinw:addmatrix:WrongInput');
+            testCase.assertTrue(isempty(testCase.swobj.matrix.mat));
+        end
+        
+        function test_single_value_adds_diag_matrix_no_label_color(testCase)
+            J = 1.0;
+            testCase.swobj.addmatrix('value', J);
+            % check diagonal elements are equal to J
+            testCase.assertTrue(isdiag(testCase.swobj.matrix.mat));
+            for elem = diag(testCase.swobj.matrix.mat)'
+               testCase.assertEqual(elem, J) 
+            end
+            % check default label and color 3 vector (randomly assigned)
+            testCase.assertEqual(testCase.swobj.matrix.label{1}, 'mat1')
+            testCase.assertEqual(size(testCase.swobj.matrix.color), [3,1])
+        end
+        
+        function test_matrix_value_added_no_modification(testCase)
+            matrix = reshape(1:9, 3, 3);
+            testCase.swobj.addmatrix('value', matrix);
+            % check diagonal elements are equal to J
+            testCase.assertEqual(testCase.swobj.matrix.mat, matrix);
```

Comment:
The comment here seems to not relate to the code below it...

Reply:
Good spot I'll remove it!